Source/WebCore/css/query/MediaQueryEvaluator.cpp

```diff
@@ -81,16 +79,7 @@ bool MediaQueryEvaluator::evaluate(const MediaQuery& query) const
         if (!document->view() || !document->documentElement())
             return EvaluationResult::Unknown;
 
-        auto defaultStyle = RenderStyle::create();
-        auto size = Style::fontSizeForKeyword(CSSValueMedium, false, *document);
-        if (size != defaultStyle.fontDescription().specifiedSize()) {
-            auto fontDescription = defaultStyle.fontDescription();
-            fontDescription.setSpecifiedSize(size);
-            fontDescription.setComputedSize(size);
-            defaultStyle.setFontDescription(WTFMove(fontDescription));
-        }
-
-        FeatureEvaluationContext context { *document, { *m_rootElementStyle, &defaultStyle, nullptr, document->renderView() }, nullptr };
+        FeatureEvaluationContext context { *document, { *m_rootElementStyle, &document->mediaQueryDefaultStyle(), nullptr, document->renderView() }, nullptr };
```

Comment:
@darinadler pointed out that no tests fail if you simply pass a nullptr instead of `&defaultStyle` here (`CSSToLengthConversionData::m_rootStyle`). I think we could either do that, or even better pass `m_rootElementStyle` as the second parameter as well. The `m_rootElementStyle` is the actual root element style that we want to resolve root relative units (like `rem`) against. 

(nullptr works because the unit resolution falls back to the main style if the root style field is null)

Reply:
@anttijk `m_rootElementStyle` doesn't seem like what you want, that causes cycles, right? As in, if a media query evaluated based on `m_rootElementStyle` but then contained `:root { font-size: ... }` inside it'd trivially be cyclic.

Reply:
@Emilio I don't think so? During style resolution `m_rootElementStyle` is not actually the root element style but a special style build for the root element by matching the UA sheet only. Thus user style can't cause cycles.

Reply:
@emilio I suppose that wasn't exactly accurate!

Reply:
> Not to constantly beat the “tests” drum but:
> 
> 1. If we are worried about cycles with `:root { font-size ... }` the best way to prove they don’t exist is to make sure we have a test.
> 2. If we think we need to pass `m_rootElementStyle` instead of `nullptr` to make `rem` work properly in media queries the best way to demonstrate that is also probably a test.

I think this is at least somewhat covered by the existing tests like LayoutTests/imported/w3c/web-platform-tests/css/mediaqueries/relative-units-001.html which verifies that `rem` unit in a media query is not affected by the author style.